gammapy/data/ivoa.py

```diff
@@ -0,0 +1,327 @@
+        name="em_max",
+        unit="m",
+        description="stop in spectral coordinates",
+        dtype="f8",
+        meta={"Utype": "Char.SpectralAxis.Coverage.Bounds.Limits.HiLimit"},
+    )
+    obscore_default[24] = Column(
+        name="em_res_power",
+        unit="",
+        description="Value of the resolving power along the spectral axis(R)",
+        dtype="f8",
+        meta={"Utype": "Char.SpectralAxis.Resolution.ResolPower.refVal"},
+    )
+    obscore_default[25] = Column(
+        name="o_ucd",
+        unit="",
+        description="Nature of the observable axis",
+        dtype="U30",
+        meta={"Utype": "Char.ObservableAxis.ucd"},
+    )
+    obscore_default[26] = Column(
+        name="pol_xel",
+        unit="",
+        description="Number of elements along the polarization axis",
+        dtype="i4",
+        meta={"Utype": "Char.PolarizationAxis.numBins"},
+    )
+    obscore_default[27] = Column(
+        name="facility_name",
+        unit="",
+        description="The name of the facility, telescope space craft used for the observation",
+        dtype="U10",
+        meta={"Utype": "Provenance.ObsConfig.Facility.name"},
+    )
+    obscore_default[28] = Column(
+        name="instrument_name",
+        unit="",
+        description="The name of the instrument used for the observation",
+        dtype="U25",
+        meta={"Utype": "Provenance.ObsConfig.Instrument.name"},
+    )
+    tab_default = Table()
+    for var in obscore_default:
+        tab_default.add_column(var)
+    return tab_default
+
+
+def _obscore_row(base_dir, single_obsID, obs_publisher_did, access_url, table):
+    """Generates an obscore row corresponding to a single obsID
+
+    Parameters
+    ----------
+    single_obsID : int
+        single Observation ID
+    **kwargs : `str` {obs_publisher_did, access_url}
+    Giving the values for is highly recommended.
+    If any of these are not given the corresponding obscore field is left empty and a warning is raised for each empty value.
+
+    Returns
+    -------
+    table : `~astropy.table.Table`
+    """
+    base_dir = make_path(base_dir)
+    data_store = DataStore.from_dir(base_dir)
+
+    if obs_publisher_did is None:
+
+        log.warning(
+            "Insufficient publisher information: 'obs_publisher_did' obscore value will be empty."
+        )
+    if access_url is None:
+        log.warning(
+            "Insufficient publisher information: access_url' obscore value will be empty."
+        )
+
+    tab = table
+    observation = data_store.obs(single_obsID)
+    obs_mask = data_store.obs_table["OBS_ID"] == observation.obs_id
+    obs_pos = data_store.obs_table[obs_mask]["EVENTS_FILENAME"]
+    path = make_path(os.environ["GAMMAPY_DATA"] + "/" + obs_pos[0])
+    size = int(os.path.getsize(path) / 1000.0)
+    tab.add_row(
+        {
+            "dataproduct_type": observation.obs_info["EXTNAME"],
+            "calib_level": 2,  # Look into the data
```

Comment:
This would be DL3 data, no? Calib level 2

Reply:
`calib_level` is a VO terminology that is not associated to our TeV one

Reply:
Hi @AtreyeeS, thank you very much for responding and sorry for my delay!  I see that @bkhelifi  has answered your first two questions well and I agree with his answers. Just a remark in general terms, yes, this tool has to be as general as possible for different types of users who need to generate both an obscore table for the entire data set as well as a subset of that data.

Regarding the other two questions:
3) We cannot use it because it is not a "standard".
4) Yes exactly, the idea is to leave them blank because they are mandatory columns in obscore, then we need to keep them anyway.